### PR TITLE
docs(readme): added types in readme

### PR DESCRIPTION
**Pull request by harish-sethuraman**
This PR contains a:

- [ ] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [x] **metadata update**

### Motivation / Use-Case
Added type definitions in readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,14 +136,19 @@ module.exports = {
 
 ### `algorithm`
 
-Type: `String|Function`
+Type:
+
+```ts
+type algorithm = string | Function;
```
Comment (alexander-akait) [inline README.md:142]: Can we show `Function`, i.e. arguments and returned type?

Comment (harish-sethuraman) [reply README.md:142]: But this function also returns a function (function that compresses)  right? We would be able to add types for its params alone?

Comment (alexander-akait) [reply README.md:142]: Here types for this function https://github.com/webpack-contrib/compression-webpack-plugin/blob/master/src/index.js#L197

**Review by alexander-akait — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (harish-sethuraman) [reply README.md]: https://github.com/webpack-contrib/compression-webpack-plugin/blob/2dd0f41cf8231247f5d9fa70df0c68011d510b56/types/index.d.ts#L172 I've added this type...

**Pull request merged**